core/NameRef.h

```diff
@@ -52,9 +52,9 @@ class NameRef final : private DebugOnlyCheck<NameRefDebugCheck> {
     // generated by tools/generate_names.cc
     struct WellKnown {};
 
-    constexpr NameRef(WellKnown, unsigned int id) : _id(id) {}
+    constexpr NameRef(WellKnown, u4 id) : _id(id) {}
```

Comment:
```suggestion
    constexpr NameRef(WellKnown, u4 id) : _id{id} {}
```
Using braced initialization here means it would be harder for us to end up in a situation like we had before -- numeric conversions aren't allowed.